### PR TITLE
feat(conformance): bundle Dapr component YAMLs into the SDK wheel

### PR DESCRIPTION
**Pull request by cmgrote**
## Summary
- Bundle `components/*.yaml` into the `atlan-application-sdk` wheel at `application_sdk/components/` (via `[tool.hatch.build.targets.wheel.force-include]`), so consumer apps get the Dapr component YAMLs from their existing `application-sdk` dependency instead of downloading them from GitHub at build time.
- Document the replacement pattern for app repos in `docs/standards/build-security.md`.
- Add conformance rule **D009 `RemoteDaprComponentFetch`** (BLOCK-tier, autofixable) to detect the network-fetch pattern in `[tool.poe.tasks.*]` and route it through the existing D-series remediation loop.

### Context
App repos (e.g. `atlan-mysql-app`) curl Dapr component YAMLs from `raw.githubusercontent.com`/the GitHub contents API, pinned to a hardcoded SDK tag (e.g. `v3.14.0`). This hits GitHub's unauthenticated rate limit under CI concurrency (429s breaking builds) and drifts from whatever `application-sdk` version is actually locked in the app's `uv.lock`. Since the SDK wheel now bundles these files, apps can copy them locally instead — no network call, and always in sync with the installed SDK version.

## Test plan
- [x] `uv build --wheel` and verified `application_sdk/components/*.yaml` are present in the built wheel
- [x] `uv run pre-commit run --all-files` — clean
- [x] `uv run python -m pytest tests/unit -x -q` — 5378 passed
- [x] `packages/conformance`: `uv run pytest tests/` — 1663 passed, 1 xfailed
- [x] `uv run atlan-application-sdk-conformance gen-rule-docs --check` — up-to-date
- [x] End-to-end: ran `atlan-application-sdk-conformance detect --series D` against a synthetic pyproject.toml mirroring `atlan-mysql-app`'s actual `download-components` task — D009 fires as a BLOCK finding at the correct line

### CODE REVIEW

File: packages/conformance/conformance/suite/checks/dependency_conformance.py [hunk 1]
```diff
@@ -849,6 +890,46 @@ def scan_text(
...
+                findings.append(
+                    _make_finding(
+                        rule_id=RULE_D009,
+                        file=file,
+                        line=ln,
+                        column=1,
+                        message=(
+                            f"poe task '{task_name}' fetches Dapr component "
+                            f"YAMLs from GitHub over the network instead of "
+                            f"reading them from the installed "
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/conformance/suite/checks/dependency_conformance.py]: Fixed in 0d74dd8: rewrote the finding loop to decide applicability across all tasks once, then report each offending line once with a task-agnostic message — no more re-scanning the whole file per task or attributing a line to the wrong task name.

File: packages/conformance/conformance/suite/checks/dependency_conformance.py [hunk 2]
```diff
@@ -62,6 +67,17 @@
...
 PYRIGHT_WEAK_MODES = frozenset({"off", "basic"})
 
+# Matches a poe task fetching Dapr component YAMLs straight from GitHub
+# (raw.githubusercontent.com or the contents API) for atlanhq/application-sdk,
+# instead of copying them from the installed wheel, which bundles them at
+# application_sdk/components/ (D009). Unauthenticated GitHub requests hit rate
+# limits under CI concurrency, and a hardcoded ref drifts from whatever SDK
+# version is actually locked in the app's uv.lock.
+_REMOTE_COMPONENT_FETCH_RE = re.compile(
+    r"(?:raw\.githubusercontent\.com|api\.github\.com)[^\s\"'\\]*/atlanhq/application-sdk\b",
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/conformance/suite/checks/dependency_conformance.py]: Fixed in 0d74dd8: replaced the trailing `\\b` with a negative lookahead `(?![\\w-])`, so `application-sdk-extra` (or any repo name that merely starts with the prefix) no longer matches. Added a regression test.

File: docs/standards/build-security.md [hunk 3]
```diff
@@ -6,8 +6,28 @@
 
 - **Image base**: `cgr.dev/chainguard-private/python` -> golden images -> SDK -> apps
 - **Dapr runtime**: baked into the `app-framework-golden` base image via Chainguard Custom Assembly (0 CVEs). The container's daprd version is owned by the Custom Assembly config — the Dockerfile no longer installs it. (The `__dapr_version` pin in `application_sdk/version.py` governs only the local-dev auto-download, a separate path.)
+- **Dapr component YAMLs**: shipped inside the `atlan-application-sdk` wheel at `application_sdk/components/` (see `[tool.hatch.build.targets.wheel.force-include]` in `pyproject.toml`, mirroring this repo's own `components/` dir). Consumer apps get them for free from their existing `application-sdk` dependency — no network download needed, and they're always in sync with whatever SDK version is locked in that app's `uv.lock`.
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply docs/standards/build-security.md]: Fixed in 0d74dd8: now consistently uses the published `atlan-application-sdk` distribution name here.

File: docs/standards/build-security.md [hunk 4]
```diff
@@ -6,8 +6,28 @@
...
 - **Registries**: Harbor (`registry.atlan.com`) for production, GHCR for CI
 
+## Consuming Dapr components in an app repo
+
+App repos should **not** curl these files from `raw.githubusercontent.com` or the GitHub contents API pinned to a hardcoded SDK tag (that pattern hits GitHub's unauthenticated rate limit under CI concurrency and silently drifts from the app's actual `application-sdk` version). Instead, copy them out of the installed package, e.g. as the app's `download-components` poe task:
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply docs/standards/build-security.md]: Fixed in 0d74dd8: updated to `atlan-application-sdk`.

File: docs/standards/build-security.md [hunk 5]
```diff
@@ -6,8 +6,28 @@
...
+download-components.shell = """
+python -c "
+import application_sdk, pathlib, shutil
+src = pathlib.Path(application_sdk.__file__).parent / 'components'
+shutil.copytree(src, 'components', dirs_exist_ok=True)
+"
+"""
+```
+
+This requires `application-sdk` to already be installed into the venv before the task runs (true for both local dev and the Docker build, where `uv sync` happens before `poe download-components`).
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply docs/standards/build-security.md]: Fixed in 0d74dd8: updated to `atlan-application-sdk`.

File: packages/conformance/tests/test_dependency_conformance.py [hunk 6]
```diff
@@ -678,6 +678,164 @@
...
+        text,
+        "pyproject.toml",
+        sdk_managed_packages=set(),
+        sdk_published_extras=set(),
+    )
+    assert len(findings) == 1
+    expected_line = next(
+        ln
+        for ln, line in enumerate(text.splitlines(), start=1)
+        if "raw.githubusercontent.com" in line
```
[the remark this answers is not in the document]
Comment (cmgrote) [reply packages/conformance/tests/test_dependency_conformance.py]: Genuine false positive as a security finding — this is a test helper locating the line number of a hardcoded fixture string the test itself constructs a few lines above (for building a test assertion), not a security/trust decision over untrusted input. That said, the substring idiom was sloppier than necessary, so fixed in 1c5738e anyway: the test now reuses `_REMOTE_COMPONENT_FETCH_RE` (the actual D009 detector regex) instead of a raw `in` check, which both satisfies CodeQL and makes the test locate the exact line the production check would flag.

**Comment by cmgrote** [comment]
@sdk-review

**Comment by cmgrote** [comment]
## SDK Review (interactive): PR #2531 — feat(conformance): bundle Dapr component YAMLs into the SDK wheel

### Verdict: READY TO MERGE

> This PR fixes a real root cause (apps curling Dapr component YAMLs from GitHub at build time, hitting rate limits and drifting from the locked SDK version) rather than patching a symptom: it bundles the files into the wheel *and* adds a BLOCK-tier conformance rule (D009) so the old pattern can't creep back in fleet-wide. The implementation is careful about line-attribution (multi-task pyproject files), false-positive avoidance (`application-sdk-extra` doesn't match), and self-check exemption. Test coverage for D009 is thorough (9 targeted cases). No guardrail violations found.

---

### Findings

No Critical or Important findings.

**`packages/conformance/conformance/suite/checks/dependency_conformance.py`**
- **Nit** [BUG] L216-252 — The D009 line-anchoring re-scans *every* line of the file for the regex once the structural gate trips (rather than only lines within the matched poe task's string value). If a pyproject.toml ever contained a stray comment elsewhere matching the same URL pattern (outside any poe task), it would also get flagged at that line. Verified this doesn't false-positive against the actual test suite or this repo's own `pyproject.toml` — very narrow edge case. *Path: optional — anchor the re-scan to the specific matched task's line range instead of the whole file, if it's ever observed in practice.*

### Holistic Recommendations
- None — root cause fix (bundle + enforce), not a symptom patch.

### Strengths
- D009 regex correctly avoids false-positiving on `application-sdk-extra`-style repo names via the negative lookahead, and correctly matches the bare-repo-reference edge case (`.../repos/atlanhq/application-sdk` with no trailing path) — both are covered by dedicated tests.
- Self-check exemption is correctly wired (`_is_self_check`) so the SDK's own local `download-components` task (`ls -l components/*.yaml`) never trips itself.
- Line-attribution test (`test_d009_multiple_tasks_one_violating_reports_once_at_correct_line`) specifically guards against misattributing the finding to the wrong task when multiple poe tasks exist.
- Docs (`build-security.md`, `dependency.md` rule catalog, `dependency.prose.md` remediation spec) are all updated consistently with the new rule, including the machine-consumed remediation prescription for the `remediate` skill.
- Verified independently: bundled `components/` files contain only placeholder/empty secret values (`aws-secrets.yaml.example`, `deployment-secrets.yaml`), nothing sensitive ships in the wheel.

---
**CI:** all applicable checks passing (the automated `sdk-review` mothership bot workflow itself errored out unrelated to PR content — this manual review was run in its place).
**Verified locally:** `uv run pytest packages/conformance/tests/test_dependency_conformance.py tests/test_catalog.py` → 126 passed.
**Models:** Claude (interactive review only — no GPT-5.3-codex adversarial pass; proxy not available in this interactive session).
**Run:** local interactive review (not a mothership-dispatched run) — no GHA run URL to link.
